baker/SiteBaker.tsx

```diff
@@ -307,17 +314,55 @@ export class SiteBaker {
         if (!this.bakeSteps.has("gdocPosts")) return
         const publishedGdocs = await Gdoc.getPublishedGdocs()
 
-        // Prefetch publishedGdocs and imageMetadata instead of each instance fetching
+        // Prefetch publishedGdocs, imageMetadata, and linkedCharts instead of each instance fetching
         const publishedGdocsDictionary = keyBy(publishedGdocs.map(clone), "id")
         const imageMetadataDictionary = await Image.find().then((images) =>
             keyBy(images, "filename")
         )
+        const publishedExplorersBySlug = await this.explorerAdminServer
+            .getAllPublishedExplorersBySlugCached()
+            .then((results) =>
+                Object.values(results).reduce(
+                    (acc, cur) => ({
+                        ...acc,
+                        [cur.slug]: {
+                            originalSlug: cur.slug,
+                            resolvedUrl: `${BAKED_BASE_URL}/${EXPLORERS_ROUTE_FOLDER}/${cur.slug}`,
+                            title: cur.title || "",
+                            thumbnail:
+                                cur.thumbnail ||
+                                `${BAKED_BASE_URL}/default-thumbnail.jpg`,
+                        },
+                    }),
+                    {} as Record<string, LinkedChart>
+                )
+            )
+        // Includes redirects
+        const publishedChartsBySlug = await Chart.mapSlugsToConfigs().then(
+            (results) =>
+                results.reduce(
+                    (acc, cur) => ({
+                        ...acc,
+                        [cur.slug]: {
+                            originalSlug: cur.slug,
+                            resolvedUrl: `${BAKED_GRAPHER_URL}/${cur.config.slug}`,
+                            title: cur.config.title || "",
+                            thumbnail: `${BAKED_GRAPHER_EXPORTS_BASE_URL}/${cur.slug}.svg`,
+                        },
+                    }),
+                    {} as Record<string, LinkedChart>
+                )
+        )
 
         for (const publishedGdoc of publishedGdocs) {
             publishedGdoc.imageMetadata = imageMetadataDictionary
             publishedGdoc.linkedDocuments = publishedGdocsDictionary
-            const publishedExplorersBySlug =
-                await this.explorerAdminServer.getAllPublishedExplorersBySlugCached()
+            publishedGdoc.linkedCharts = {
+                ...publishedChartsBySlug,
+                ...publishedExplorersBySlug,
+            }
+            // this is a no-op if the gdoc doesn't have an all-chart block
+            await publishedGdoc.loadRelatedCharts()
```

Comment:
This is from #2326 , where I made this commit initially but then cherry-picked over to a new branch to isolate it for a PR.

It fixes a bug so it's fine to merge it here too.